### PR TITLE
[inductor] Support alpha= in add/sub

### LINKED ISSUE #1085: Inductor fails on optimizer.step
Repro:
```
import torch
import torch.optim as optim
import torchdynamo

torchdynamo.config.fake_tensor_propagation = False

input = torch.ones([10, 10])
model = torch.nn.Sequential(*[torch.nn.Linear(10, 10) for _ in range(2)])
opt_model = torchdynamo.optimize("inductor")(model)
opt_model(input).sum().backward()

optimizer = optim.SGD(opt_model.parameters(), lr=5e-5)
opt_fn = torchdynamo.optimize("inductor")(optimizer.step)
opt_fn()
```

Error:
```
Traceback (most recent call last):
  File "/data/home/ybliang/Meta/torchdynamo/torchinductor/graph.py", line 234, in call_function
    return lowerings[target](*args, **kwargs)
  File "/data/home/ybliang/Meta/torchdynamo/torchinductor/lowering.py", line 185, in wrapped
    return decomp_fn(*args, **kwargs)
TypeError: fn() got an unexpected keyword argument 'alpha'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/data/home/ybliang/Meta/torchdynamo/debug/debug3.py", line 14, in <module>
    opt_fn()
  File "/data/home/ybliang/Meta/torchdynamo/torchdynamo/eval_frame.py", line 154, in _fn
    return fn(*args, **kwargs)
  File "/fsx/users/ybliang/conda/envs/dynamo-gpu/lib/python3.9/site-packages/torch/optim/optimizer.py", line 23, in _use_grad
    ret = func(self, *args, **kwargs)
  File "/fsx/users/ybliang/conda/envs/dynamo-gpu/lib/python3.9/site-packages/torch/optim/sgd.py", line 151, in step
    sgd(params_with_grad,
  File "/fsx/users/ybliang/conda/envs/dynamo-gpu/lib/python3.9/site-packages/torch/optim/sgd.py", line 171, in sgd
    def sgd(params: List[Tensor],
  File "/data/home/ybliang/Meta/torchdynamo/torchdynamo/eval_frame.py", line 154, in _fn
    return fn(*args, **kwargs)
  File "/fsx/users/ybliang/conda/envs/dynamo-gpu/lib/python3.9/site-packages/functorch/_src/aot_autograd.py", line 886, in forward
    return compiled_f(
  File "/fsx/users/ybliang/conda/envs/dynamo-gpu/lib/python3.9/site-packages/functorch/_src/aot_autograd.py", line 872, in new_func
    compiled_fn = create_aot_dispatcher_function(
  File "/fsx/users/ybliang/conda/envs/dynamo-gpu/lib/python3.9/site-packages/functorch/_src/aot_autograd.py", line 450, in create_aot_dispatcher_function
    return aot_dispatch_base(flat_fn, fake_flat_tensor_args, aot_config)
  File "/fsx/users/ybliang/conda/envs/dynamo-gpu/lib/python3.9/site-packages/functorch/_src/aot_autograd.py", line 276, in aot_dispatch_base
    compiled_fw = aot_config.fw_compiler(fw_module, flat_args)
  File "/data/home/ybliang/Meta/torchdynamo/torchinductor/compile_fx.py", line 170, in fw_compiler
    return compile_fx_inner(
  File "/data/home/ybliang/Meta/torchdynamo/torchdynamo/debug_utils.py", line 253, in debug_wrapper
    raise e
  File "/data/home/ybliang/Meta/torchdynamo/torchdynamo/debug_utils.py", line 241, in debug_wrapper
    compiled_fn = compiler(gm, example_inputs, **kwargs)
  File "/data/home/ybliang/Meta/torchdynamo/torchinductor/debug.py", line 182, in inner
    return fn(*args, **kwargs)
  File "/data/home/ybliang/Meta/torchdynamo/torchinductor/compile_fx.py", line 56, in compile_fx_inner
    wrap(graph.run)(*example_inputs)
  File "/data/home/ybliang/Meta/torchdynamo/torchinductor/graph.py", line 129, in run
    return super().run(*args)
  File "/fsx/users/ybliang/conda/envs/dynamo-gpu/lib/python3.9/site-packages/torch/fx/interpreter.py", line 129, in run
    self.env[node] = self.run_node(node)
  File "/data/home/ybliang/Meta/torchdynamo/torchinductor/graph.py", line 285, in run_node
    result = super().run_node(n)
  File "/fsx/users/ybliang/conda/envs/dynamo-gpu/lib/python3.9/site-packages/torch/fx/interpreter.py", line 162, in run_node
    return getattr(self, n.op)(n.target, args, kwargs)
  File "/data/home/ybliang/Meta/torchdynamo/torchinductor/graph.py", line 236, in call_function
    raise LoweringException(e, target, args, kwargs) from e
torchinductor.exc.LoweringException: TypeError: fn() got an unexpected keyword argument 'alpha'
  target: aten.add_.Tensor
  args[0]: TensorBox(StorageBox(
    InputBuffer(name='arg0_1', layout=FixedLayout('cpu', torch.float32, size=[s0, s0], stride=[s0, 1]))
  ))
  args[1]: TensorBox(StorageBox(
    InputBuffer(name='arg4_1', layout=FixedLayout('cpu', torch.float32, size=[s0, s0], stride=[s0, 1]))
  ))
  kwargs: {'alpha': -5e-05}

```
This failure occurs on both CPU and cuda.

### PR DESCRIPTION
**Pull request by jansel**
Fixes #1085

### CODE REVIEW

File: torchinductor/lowering.py [hunk 1]
```diff
@@ -249,9 +249,17 @@ def promote_constants(inputs):
...
-    def inner(*inputs: List[TensorBox]):
+def make_pointwise(
+    fn, override_dtype=None, override_device=None, override_bool=None, allow_alpha=False
+):
+    def inner(*inputs: List[TensorBox], alpha=None):
         inputs = promote_constants(inputs)
+        if allow_alpha:
+            if alpha is not None and alpha != 1:
+                inputs = list(inputs)
+                inputs[-1] = mul(inputs[-1], alpha)
```
Comment (voznesenskym) [inline torchinductor/lowering.py:260]: assert alpha is a scalar?

Comment (jansel) [reply torchinductor/lowering.py:260]: `mul()` already has asserts on its inputs.

Comment (voznesenskym) [reply torchinductor/lowering.py:260]: Hmm, I assumed mul would be able to take tensors, where alpha was required to be a scalar. As long as mul asserts, sgtm!

**Review by voznesenskym — APPROVED** [review]

File: torchinductor/lowering.py (hunk 1 above)
Comment (jansel) [reply torchinductor/lowering.py:260]: Often constants get wrapped as tensors in AOT Autograd.  So we want this to work for tensors.

Comment (ngimel) [reply torchinductor/lowering.py:260]: But we still need to check that it's 0d and that it's dtype is correct for the other inputs (e.g. integral inputs and float alpha error out in eager)

**Comment by lezcano** [comment]
Why was this added here, rather than simply using the `add` decomposition from PrimTorch and let the lowering fuse the pointwise operations? The PrimTorch operation already has the correct checks on alpha.

**Comment by ngimel** [comment]
we can't use aten -> prim decompositions from primTorch because they hardcode broadcasting.

**Pull request merged**